### PR TITLE
Use lightweight filterChain cache structure as starting point

### PR DESCRIPTION
**Pull request by gjdeval**
Many Liberty apps will not have large numbers of web app filter chains, so we should start with a lighter weight filter chain cache structure than the current MRU approach

### CODE REVIEW

**Comment by wraschke** [comment]
#build

**Comment by gjdeval** [comment]
#libby

**Comment by gjdeval** [comment]
#build

**Comment by gjdeval** [comment]
#build

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/filter/WebAppFilterManager.java [hunk 1]
```diff
@@ -17,16 +17,17 @@
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 import java.text.MessageFormat;
+import java.util.AbstractMap;
```
Comment (jhanders34) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/filter/WebAppFilterManager.java:20]:
```suggestion
```

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/filter/WebAppFilterManager.java [hunk 2]
```diff
@@ -17,16 +17,17 @@
...
-import java.util.Collections;
 import java.util.EnumSet;
-import java.util.Enumeration;
-import java.util.HashMap;
-import java.util.Hashtable;
 import java.util.Iterator;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
```
Comment (jhanders34) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/filter/WebAppFilterManager.java:27]:
```suggestion
```

**Review by jhanders34 — APPROVED** [review]

**Comment by gjdeval** [comment]
#libby

**Pull request merged**